v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -0,0 +1,647 @@
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, SUBTASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, {});
+      });
+    });
+
+    describe("task and subtask with named parameters", function () {
+      before(async function () {
+        TASK = task(["task"])
+          .addNamedParameter({ name: "param" })
+          .addNamedParameter({
+            name: "flag",
+            type: ParameterType.BOOLEAN,
+          })
+          .addNamedParameter({
+            name: "optionFlag",
+          })
+          .setAction(() => {})
+          .build();
+
+        SUBTASK = task(["task", "subtask"])
+          .addNamedParameter({ name: "param" })
+          .setAction(() => {})
+          .build();
+
+        hre = await createHardhatRuntimeEnvironment({
+          tasks: [TASK, SUBTASK],
+        });
+      });
+
+      it("should get the tasks and its parameter", function () {
+        const command = "npx hardhat task --param <paramValue>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, TASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, {
+          param: "<paramValue>",
+        });
+      });
+
+      it("should get the subtask and its parameter", async function () {
+        const command = "npx hardhat task subtask --param <paramValue>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, SUBTASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, {
+          param: "<paramValue>",
+        });
+      });
+
+      it("should get the tasks and its parameter as type boolean with value set to true", function () {
+        const command = "npx hardhat task --flag";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, TASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true]);
+        assert.deepStrictEqual(res.taskArguments, { flag: true });
+      });
+
+      it("should throw because the parameter when set as bool (flag behavior) should not be followed by a bool value", function () {
```

Comment:
but this is not throwing, right? we should update the description

Reply:
[Done in this commit](https://github.com/NomicFoundation/hardhat/pull/5282/commits/d2e2f9877dfbba79626a4417708dbd2591f9217a)